site/en/blog/canvas2d/index.md

```diff
@@ -0,0 +1,282 @@
+  const height = 100;
+  const radius = 20;
+
+  ctx.beginPath();
+  ctx.moveTo(left + radius, top);
+  ctx.lineTo(left + width - radius, top);
+  ctx.arcTo(left + width, top, left + width, top + radius, radius);  
+  ctx.lineTo(left + width, top + height - radius);
+  ctx.arcTo(
+left + width, top + height, left + width - radius, 
+top + height, radius);
+  ctx.lineTo(left + radius, top + height);
+  ctx.arcTo(left, top + height, left, top + height - radius, radius)
+  ctx.lineTo(left, top + radius);
+  ctx.arcTo(left, top, left + radius, top, radius)
+  ctx.stroke();
+```
+
+All this was necessary for a modest simple rounded rectangle:
+
+{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/J9JnDFq28BhDGsMyw2zY.png", alt="A rounded rectangle.", width="300", height="150" %}
+
+With the new API there's a `roundRect()` method.
+
+```js
+	ctx.roundRect(upper, left, width, height, borderRadius);
+
+	So the above can be wholly replaced by:
+
+	ctx.roundRect(10, 10, 200, 100, 20);
+```
+
+	`ctx.roundRect()` also can take in an array for the `borderRadius` argument of up to four numbers. These radii control the four corners of the rounded rectangle the same way [as for CSS](https://developer.mozilla.org/docs/Web/CSS/border-radius). For example:
+
+```js
+	ctx.roundRect(10, 10, 200, 100, [15, 50, 30])
+```
+
+[Check out the demo to play around](https://pickle-gorgeous-treatment.glitch.me/)!
+
+### Conic Gradient
+
+You've seen linear gradients.
+
+```js
+  const gradient = ctx.createLinearGradient(0, 0, 200, 100);
+  gradient.addColorStop(0, "blue");
+  gradient.addColorStop(0.5, "magenta");
+  gradient.addColorStop(1, "white");
+  ctx.fillStyle = gradient;
+  ctx.fillRect(10, 10, 200, 100);
+```
+
+{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/rkG4JDoOC42xn7szSZbL.png", alt="A linear gradient.", width="300", height="150" %}
+
+Radial gradients.
+
+```js
+  const radialGradient = ctx.createRadialGradient(150, 75, 10, 150, 75, 70);
+  radialGradient.addColorStop(0, "white");
+  radialGradient.addColorStop(0.5, "magenta");
+  radialGradient.addColorStop(1, "lightblue");
+
+  ctx.fillStyle = radialGradient;
+  ctx.fillRect(0, 0, canvas.width, canvas.height);
+```
+
+{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/NF89ac1lvs6jb2kxTAja.png", alt="A radial gradient.", width="300", height="150" %}
+
+but how about a nice conic gradient?
+
+```js
+const grad = ctx.createConicGradient(0, 100, 100);
+
+grad.addColorStop(0, "red");
+grad.addColorStop(0.25, "orange");
+grad.addColorStop(0.5, "yellow");
+grad.addColorStop(0.75, "green");
+grad.addColorStop(1, "blue");
+
+ctx.fillStyle = grad;
+ctx.fillRect(0, 0, 200, 200);
+```
+
+{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/s05ut90JNhvLywJ9Sb9b.png", alt="A conic gradient.", width="451", height="329" %}
+
+### Text Modifiers
+
+Canvas2Ds text rendering capabilities have been woefully behind. 
+Chrome is pleased to announce the addition of a whole whack of new attributes to Canvas2D text rendering:
```

Comment:
>Chrome is pleased to announce

The definitions below are part of the HTML spec. This wording makes it sound like these are Chrome-specific. They look web platform to me.

Reply:
Yeah, maybe something like "Chrome has implemented a whole new ..."

Reply:
Sounds good.